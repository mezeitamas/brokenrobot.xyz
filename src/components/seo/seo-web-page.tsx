import React from 'react';
import type { FunctionComponent, PropsWithChildren } from 'react';

import { useSiteMetadata } from '../site-metadata/use-site-metadata';

import { MetaOgWebsite } from './meta-og-website/meta-og-website';
import { MetaTwitter } from './meta-twitter/meta-twitter';
import { RichResultsWebPage } from './rich-results-web-page/rich-results-web-page';

type SeoWebPageProps = {
    title: string;
    description: string;
    pathname: string;
};

const SeoWebPage: FunctionComponent<PropsWithChildren<SeoWebPageProps>> = ({
    title,
    description,
    pathname,
    children
}) => {
    const { title: siteName, siteUrl, author } = useSiteMetadata();
    const url = `${siteUrl}${pathname}`;

    return (
        <>
            <html lang="en" />

            <title>{title}</title>

            <meta
                httpEquiv="cache-control"
                content="public, max-age=0, must-revalidate"
            />

            <meta
                name="description"
                content={description}
            />

            <MetaOgWebsite
                siteName={siteName}
                title={title}
                description={description}
                url={url}
            />

            <RichResultsWebPage
                siteName={siteName}
                siteUrl={siteUrl}
                author={author}
            />

            <MetaTwitter
                title={title}
                description={description}
                url={url}
                author={author}
            />

            {children}
        </>
    );
};

export { SeoWebPage };

export type { SeoWebPageProps };
