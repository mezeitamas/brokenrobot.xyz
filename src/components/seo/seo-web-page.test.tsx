import React from 'react';

import { create } from 'react-test-renderer';

import { SeoWebPage } from './seo-web-page';
import type { SeoWebPageProps } from './seo-web-page';

jest.mock('../../hooks/use-site-metadata', () => {
    return {
        useSiteMetadata: jest.fn().mockImplementation(() => {
            return {
                title: 'Broken Robot',
                description:
                    'Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world.',
                siteUrl: 'https://www.brokenrobot.xyz',
                author: {
                    name: 'Tamas Mezei',
                    social: {
                        github: 'https://github.com/mezeitamas',
                        linkedin: 'https://www.linkedin.com/in/mezeitamas/'
                    }
                }
            };
        })
    };
});

describe('<SeoWebPage />', () => {
    test('should render', () => {
        // Given
        const props: SeoWebPageProps = {
            title: 'Broken Robot',
            description:
                'Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world.',
            pathname: ''
        };

        // When
        const component = create(<SeoWebPage {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
