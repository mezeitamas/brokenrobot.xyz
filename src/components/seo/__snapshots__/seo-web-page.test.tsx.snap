// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeoWebPage /> should render 1`] = `
[
  <html
    lang="en"
  />,
  <title>
    Broken Robot
  </title>,
  <meta
    content="public, max-age=0, must-revalidate"
    httpEquiv="cache-control"
  />,
  <meta
    content="Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world."
    name="description"
  />,
  <meta
    content="website"
    property="og:type"
  />,
  <meta
    content="en_US"
    property="og:locale"
  />,
  <meta
    content="Broken Robot"
    property="og:site_name"
  />,
  <meta
    content="Broken Robot"
    property="og:title"
  />,
  <meta
    content="Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world."
    property="og:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz"
    property="og:url"
  />,
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"WebPage","@id":"https://www.brokenrobot.xyz/#WebPage","name":"Broken Robot","url":"https://www.brokenrobot.xyz","inLanguage":"en-US","author":[{"@type":"Person","@id":"https://www.brokenrobot.xyz/about-me/#Person","name":"Tamas Mezei","url":"https://www.brokenrobot.xyz/about-me/","sameAs":["https://github.com/mezeitamas","https://www.linkedin.com/in/mezeitamas/","https://mastodon.social/@brokenrobot_xyz","https://twitter.com/brokenrobot_xyz"]}],"publisher":{"@type":"Organization","@id":"https://www.brokenrobot.xyz/#Organization","name":"Broken Robot","url":"https://www.brokenrobot.xyz"}}
  </script>,
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="Broken Robot"
    name="twitter:title"
  />,
  <meta
    content="Personal website and blog of Tamas Mezei. Welcome to my little corner of the web, where I share my professional experiences, thoughts, adventures, and projects with the world."
    name="twitter:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz"
    name="twitter:url"
  />,
  <meta
    content="@brokenrobot_xyz"
    name="twitter:creator"
  />,
]
`;
